plugins/m2k/src/old/signal_generator.cpp

```diff
@@ -2272,6 +2272,12 @@ size_t SignalGenerator::get_samples_count(unsigned int chnIdx, double rate, bool
 	}
 
 out_cleanup:
+
+	// for size 0 return 0 going further with this value will get stuck in an infinite loop
+	if(size == 0) {
+		return 0;
```

Comment:
I think we should disable the Run button in this case.
What happens here is the the SigGen does not output anything, but the Run button is checked, suggesting that data is in fact generated on the DAC, which is not the case..